feat(tasks): SAV-738: Make possible to skip task initialisation with explicit enabled: false

dannash100 opened the pull request:
### Changes

- Makes it possible to disable a task by explicitly setting enabled: false 
- This came out of a suggestion on my recent pr with materialised view refresh task

### Deploys

- [x] **Deploy to Tamanu Internal** <!-- #deploy -->

### Remember to...

- ...write or update tests
- ...add UI screenshots and **testing notes** to the Linear issue
- ...add any **manual upgrade steps** to the Linear issue
- ...update the [config reference](https://beyond-essential.slab.com/posts/reference-config-file-0c70ukly), [settings reference](https://beyond-essential.slab.com/posts/reference-settings-0blw1x2q), or any [relevant runbook(s)](https://beyond-essential.slab.com/topics/runbooks-bs04ml6c)
- ...call out additions or changes to **config files** for the deployment team to take note of

<!-- Thank you! -->

dannash100 committed:
nass-1260: opps

dannash100 committed:
feat: add ability to disable tasks to facility server

dannash100 committed:
feat: make work for central too

dannash100 committed:
feat: fix stuff that was copiloted lol

dannash100 committed:
remove file didnt mean to add

edmofro reviewed (changes requested):
Sorry I meant to say that I was also planning to go through the existing scheduled tasks that manage the enabled flag manually, and remove it

dannash100 committed:
changes

dannash100 committed:
Merge branch 'main' into feat/add-enabled-flag-to-scheduled-tasks

Commit:
Merge branch 'main' into feat/add-enabled-flag-to-scheduled-tasks

Commit:
SAV-738: Reworked enabled flag to be an argument passed in to the ScheduledTask constructor
- Allows for the various config schemas that different tasks use
- Allows for all kinds of ScheduledTasks to be disabled
- ScheduledTasks can handle their own disabled behaviour

edmofro reviewed (approved):
Yep this is much better - doesn't add any more assumptions and works across the board. Honestly the first pass was just a quick win because I had a few minutes and thought it was insane that `enabled` wasn't universally supported, but this solution is a big improvement on that.

Commit:
SAV-738: Added unit tests for ScheduledTask

Klaus-S committed:
Merge branch 'main' into feat/add-enabled-flag-to-scheduled-tasks
